### DISCUSSION TITLE
Testing notes for the Sony A7M4

### DISCUSSION THREAD

**Discussion by ik-ne** [frank26080115/alpha-fairy#16] [General]
So I’ve been playing with this for an evening or two (after I found an M5StickC-Plus actually in stock for not silly prices) with my A7IV and just wanted to put up some very quick notes about what I have found to work or not work so far.

Wi-Fi connection through “Wi-Fi Access Point” doesn’t work. Connects then disconnects. Possibly due to needing the ‘Handshake’ data from packet-sniffing.

Connecting through “Wi-Fi Direct” however works just fine, use the Auto-Connect function on the M5StickC and it just works.

What functions do and don’t work;

Remote;
* Quick Remote - Working as intended
* Timed Remote Shutter - Working as intended
* Shutter Trigger - Working as intended
* Dual Shutter - As stated in the docs this is finicky. Sometimes it worked, most of the time however it just didn’t. It definitely tries too. But fails somewhere along the way
* Infrared Timecode Reset - Sony removed the IR functionality from the A7IV so unable to test

Focus;
*edit - I had some more time to sit down with this and so this section gets an overhaul from my initial tests after I found out I am mostly an idiot;*
* Focus Stack - Working as intended 
* Focus 9 Point - Working as intended
* Shutter Speed Step - Working as intended. Found one caveat with this one however, you need to make sure that you are in 'MF' *before* you use this. If you let the remote change you from 'AF' to 'MF' it locks you into MF and needs a camera restart to get you out of it. 
* Focus Pull - Works as intended

Intervalometer;
* From my very quick testing with this, everything works as intended. The only thing to note (as said in the docs) the battery on the M5StackC really isn’t that great so be mindful of that
* Although I haven’t tested the Astro-Intervalometer specifically, there should be no reason this doesn’t work as above

Virtual Top Panel;
* Working as intended

Tally Light;
* Working as intended 

Focus Frustration:
* Not tested

As I said, this is just from the past evening or two with some spare time after work, but all of the "standard" remote-type functions seem to work absolutely fine which is what the majority of my use will revolve around anyway.

My thanks to @frank26080115, this really is a nifty little project and somehow manages to be more efficient and useful than Sony's own mostly dreadful phone app.

**Comment by frank26080115** [comment]
Thanks! Glad to hear it works well and you like it!

**Comment by Arbel-arad** [comment]
it appears to be the same case about the connection methods on my a7rV,
but i keep getting a wifi error after it is connected for a few minutes.
there is also about a 1-2 second delay between a button press and response when the device becomes unresponsive until the camera receives the command.

my plan is to make a small device that will fit on the hot shoe mount and the empty left side (where the a1 has another dial) with a dial plus some buttons for quick focus adjustments and HDR stacks, while also being able to take it off to use as a remote for astrophotography.

this will probably be made of a [seeed xiao](https://www.seeedstudio.com/XIAO-ESP32S3-p-5627.html) on a custom PCB with an encoder dial.
maybe it can even be charged from the connector (from what i found there are power connections but i don't know where to get a plug for it)

**Comment by Interceptor121** [comment]
> So I’ve been playing with this for an evening or two (after I found an M5StickC-Plus actually in stock for not silly prices) with my A7IV and just wanted to put up some very quick notes about what I have found to work or not work so far.
> 
> Wi-Fi connection through “Wi-Fi Access Point” doesn’t work. Connects then disconnects. Possibly due to needing the ‘Handshake’ data from packet-sniffing.
> 
> Connecting through “Wi-Fi Direct” however works just fine, use the Auto-Connect function on the M5StickC and it just works.
> 
> What functions do and don’t work;
> 
> Remote;
> 
> * Quick Remote - Working as intended
> * Timed Remote Shutter - Working as intended
> * Shutter Trigger - Working as intended
> * Dual Shutter - As stated in the docs this is finicky. Sometimes it worked, most of the time however it just didn’t. It definitely tries too. But fails somewhere along the way
> * Infrared Timecode Reset - Sony removed the IR functionality from the A7IV so unable to test
> 
> Focus; _edit - I had some more time to sit down with this and so this section gets an overhaul from my initial tests after I found out I am mostly an idiot;_
> 
> * Focus Stack - Working as intended
> * Focus 9 Point - Working as intended
> * Shutter Speed Step - Working as intended. Found one caveat with this one however, you need to make sure that you are in 'MF' _before_ you use this. If you let the remote change you from 'AF' to 'MF' it locks you into MF and needs a camera restart to get you out of it.
> * Focus Pull - Works as intended
> 
> Intervalometer;
> 
> * From my very quick testing with this, everything works as intended. The only thing to note (as said in the docs) the battery on the M5StackC really isn’t that great so be mindful of that
> * Although I haven’t tested the Astro-Intervalometer specifically, there should be no reason this doesn’t work as above
> 
> Virtual Top Panel;
> 
> * Working as intended
> 
> Tally Light;
> 
> * Working as intended
> 
> Focus Frustration:
> 
> * Not tested
> 
> As I said, this is just from the past evening or two with some spare time after work, but all of the "standard" remote-type functions seem to work absolutely fine which is what the majority of my use will revolve around anyway.
> 
> My thanks to @frank26080115, this really is a nifty little project and somehow manages to be more efficient and useful than Sony's own mostly dreadful phone app.

Re focus stacking I have now tried several mobile apps that use the same SDK with different results
One app has a choice between small medium and large step appears to work on my A1 takes the shots relatively quickly but then when you examine the focus did not shift any of the shots taken
The second one has a different type of operation it asks you to set initial and final position and then determines the steps. This occasionally works but other times goes into an endless loop. In order to move the focus it wiggles the control until it reaches balance and then shoots, which means it takes 3-4 second for each shot

Have you checked that the focus position has actually moved on screen when you have done your tests? 
For me understanding and setting the steps and the number of shots required is essential, if you just shoot endlessly from 0 to infinity and you don't even know how the camera is moving the focus this is not useful for real life scenarios
